client-spark/common/src/main/java/org/apache/spark/shuffle/handle/MutableShuffleHandleInfo.java

```diff
@@ -98,18 +98,18 @@ private Map<Integer, Map<Integer, List<ShuffleServerInfo>>> toPartitionReplicaMa
     return partitionReplicaAssignedServers;
   }
 
-  public Set<ShuffleServerInfo> getReplacements(String faultyServerId) {
+  public List<ShuffleServerInfo> getReplacements(String faultyServerId) {
     return excludedServerToReplacements.get(faultyServerId);
   }
 
-  public Set<ShuffleServerInfo> updateAssignment(
-      int partitionId, String receivingFailureServerId, Set<ShuffleServerInfo> replacements) {
+  public List<ShuffleServerInfo> updateAssignment(
+      int partitionId, String receivingFailureServerId, List<ShuffleServerInfo> replacements) {
     if (replacements == null || StringUtils.isEmpty(receivingFailureServerId)) {
-      return Collections.emptySet();
+      return Collections.emptyList();
     }
     excludedServerToReplacements.put(receivingFailureServerId, replacements);
```

Comment:
@zuston I found there are something wrong with `excludedServerToReplacements` for split partition, as the server is not faulty server only just cannot store more data for specific partition, so it may should not be added to the exclude servers.

Reference to 

```java
  public Set<String> listExcludedServers() {
    return excludedServerToReplacements.keySet();
  }

```